src/ast/implementation/declaration/variable_declaration.ts

```diff
@@ -158,33 +158,60 @@ export class VariableDeclaration extends ASTNode {
         const type = this.vType;
 
         if (type instanceof UserDefinedTypeName) {
-            const declaration = type.vReferencedDeclaration;
+            const site = this.getClosestParentByType(ContractDefinition);
 
-            if (declaration instanceof StructDefinition) {
-                const signatures = declaration.vMembers.map(
-                    (member) => member.canonicalSignatureType
+            if (site === undefined) {
+                throw new Error(
+                    `Unable to compute canonical signature type for variables outside of contract: ${this.print()}`
                 );
-
-                return "(" + signatures.join(",") + ")";
             }
 
-            if (declaration instanceof ContractDefinition) {
-                return "address";
-            }
+            const declaration = type.vReferencedDeclaration;
 
-            if (declaration instanceof EnumDefinition) {
-                const length = declaration.children.length;
+            if (site.kind === ContractKind.Library) {
+                if (declaration instanceof ContractDefinition) {
+                    return declaration.name;
+                }
 
-                for (let n = 8; n <= 32; n += 8) {
-                    if (length < 2 ** n) {
-                        return "uint" + n;
-                    }
+                if (
+                    declaration instanceof StructDefinition ||
+                    declaration instanceof EnumDefinition
+                ) {
+                    return declaration.vScope instanceof ContractDefinition
```

Comment:
You can use `getUserDefinedTypeFQName` from `utils.ts` here.

Reply:
Will update.

Reply:
Fixed in https://github.com/ConsenSys/solc-typed-ast/pull/51/commits/08fe8e8b015fcfea89c825ef869fc18a7aa92d58